service/src/main/java/uk/gov/hmcts/reform/fpl/service/CreateC21OrderService.java

```diff
@@ -0,0 +1,95 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.google.common.collect.ImmutableMap;
+import org.assertj.core.util.Lists;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.config.HmctsCourtLookupConfiguration;
+import uk.gov.hmcts.reform.fpl.model.C21Order;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.common.C21OrderBundle;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.JudgeAndLegalAdvisor;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+import uk.gov.hmcts.reform.fpl.utils.JudgeAndLegalAdvisorHelper;
+
+import java.time.format.FormatStyle;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang3.ObjectUtils.defaultIfNull;
+
+@Service
+public class CreateC21OrderService {
+
+    private final DateFormatterService dateFormatterService;
+    private final HmctsCourtLookupConfiguration hmctsCourtLookupConfiguration;
+    private final Time time;
+
+    public CreateC21OrderService(DateFormatterService dateFormatterService,
+                                 HmctsCourtLookupConfiguration hmctsCourtLookupConfiguration, Time time) {
+        this.dateFormatterService = dateFormatterService;
+        this.hmctsCourtLookupConfiguration = hmctsCourtLookupConfiguration;
+        this.time = time;
+    }
+
+    public Map<String, Object> getC21OrderTemplateData(CaseData caseData) {
+        return ImmutableMap.<String, Object>builder()
+            .put("familyManCaseNumber", caseData.getFamilyManCaseNumber())
+            .put("courtName", getCourtName(caseData.getCaseLocalAuthority()))
+            .put("orderTitle", getOrderTitle(caseData))
+            .put("orderDetails", caseData.getTemporaryC21Order().getOrderDetails())
+            .put("todaysDate", dateFormatterService.formatLocalDateToString(time.now().toLocalDate(), FormatStyle.LONG))
```

Comment:
hmm I am not convinced on this time thing 🤔

Reply:
Don't see too much of a problem... it works, no? testing works too... (ignoring midnight). Adam can check this too, he asked for us to look into the date